Src/FluentAssertions/CallerIdentification/MultiLineCommentParsingStrategy.cs

```diff
@@ -26,10 +26,9 @@ public ParsingState Parse(char symbol, StringBuilder statement)
             return ParsingState.GoToNextSymbol;
         }
 
-        var isStartOfMultilineComment =
-            symbol is '*'
-            && statement.Length > 0
-            && statement[^1] is '/';
+#pragma warning disable SA1010 // https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/3507
+        var isStartOfMultilineComment = symbol is '*' && statement is [.., '/'];
+#pragma warning restore SA1010
```

Comment:
Difference: 
A null check of `statement`

[SharpLab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABADAAgwRgDoAVGBAFwG4BYAKDuAggBscAFAQygGcYB5ZgBMAFGAAWXHNwCeAW0bMANDgDK5KAEsAdgHMAQgFcNQmFCnkO5GLJhbyASjoBvOjjc4AbpI3c1XcnwAZgCyBszkxtowAMIQsjZ2OAC8ru5paTLyLDg+OADkAFR5qenpAGRl5pbWtuSEADK2OuRiOAB8OFglpe4VVVYJ5ADaAHr4ALo53PkA9Hk0tGndGADsU35QASFhEcxRsfG1CwC+dAxMrJw8MAByMADuohJmmQrKapq6hsaCpv01dkctBci3cXjMPg2W1C4UiWhicUGySkcgUU3yRRwfW4FgGtXRQ0IhGUeTm4wWS1BblW6wsmyCMN2+0RRzoxyAA=)

![image](https://user-images.githubusercontent.com/919634/229340785-ead14038-3f3d-4ace-b33d-65b04a2ebdd3.png)